pkg/migration/azure.go

```diff
@@ -259,5 +265,67 @@ func (m *azureMigrator) updateCRs(ctx context.Context) error {
 // triggerMigration executes the last missing updates on CRs so that
 // reconciliation transistions to upstream controllers.
 func (m *azureMigrator) triggerMigration(ctx context.Context) error {
+	releaseComponents := getReleaseComponents(m.crs.release)
+
+	{
```

Comment:
Disclaimer: I have no idea what I'm doing...

- I don't know if I included all necessary CRs. I tried to reason the reconciliation order, but I have a nagging feeling that something is still missing. We'll find out.
- I have no clue if this order is right or if it actually matters in the end. I know that when supplying CRs in one big YAML, the order matters because otherwise things have changed in the middle of `apply`.
- The distinct CAPI components make absolutely no sense to me. Why one kubeadm CR is `bootstrap` while the other one (doing also function of bootstrap) is `control-plane` - this is just confusing and error prone.

Nevertheless, at least this is some kind of start and we can then iterate once we get to testing. :man_shrugging: